client/layout/masterbar/quick-language-switcher.jsx

```diff
@@ -0,0 +1,59 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { getLocaleSlug } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+
+import MasterbarItem from './item';
+import LanguagePickerModal from 'components/language-picker/modal';
+import switchLocale from 'lib/i18n-utils/switch-locale';
+import config from 'config';
+
+class QuickLanguageSwitcher extends React.Component {
+	state = {
+		isShowingModal: false,
+	};
+
+	toggleLanguagesModal = () => {
+		this.setState( state => ( {
+			isShowingModal: ! state.isShowingModal,
+		} ) );
+	};
+
+	onClick = event => {
+		this.toggleLanguagesModal();
+		event.preventDefault();
+		return;
+	};
+
+	onSelected = languageSlug => {
+		switchLocale( languageSlug );
+	};
+
+	render() {
+		const selectedLanguageSlug = getLocaleSlug();
+
+		return (
+			<div className="masterbar__item quick-language-switcher">
```

Comment:
I think the convention for CSS class names is `dir__component`, i.e., `masterbar__quick-language-switcher`

This will also remove the double color onHover.